AngularApp/src/app/diagnostic-data/services/telemetry/telemetry.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable, OnInit, Inject } from '@angular/core';
+import { TelemetryEventNames, ITelemetryProvider } from './telemetry.common';
+import { DIAGNOSTIC_DATA_CONFIG, DiagnosticDataConfig } from '../../config/diagnostic-data-config';
+import { AppInsightsTelemetryService } from './appinsights-telemetry.service';
+import { KustoTelemetryService } from './kusto-telemetry.service';
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable()
+export class TelemetryService {
+    private telemetryProviders: ITelemetryProvider[] = [];
+    private commonDetectorEventProperties: { [name: string]: string };
+    eventPropertiesSubject: BehaviorSubject<any> = new BehaviorSubject<any>(null);
+    eventPropertiesLocalCopy: { [name: string]: string };
+
+    constructor(private _appInsightsService: AppInsightsTelemetryService, private _kustoService: KustoTelemetryService, @Inject(DIAGNOSTIC_DATA_CONFIG) private config: DiagnosticDataConfig) {
+        if (config.useKustoForTelemetry) {
+            this.telemetryProviders.push(this._kustoService);
+        }
+        if (config.useAppInsightsForTelemetry) {
+            this.telemetryProviders.push(this._appInsightsService);
+        }
+
+        this.eventPropertiesSubject.subscribe((data: any) => {
+            this.eventPropertiesLocalCopy = data;
+        });
+    }
+
+    public setDetectEventProperties(eventProperties: { [name: string]: string }) {
+        this.commonDetectorEventProperties = eventProperties;
+    }
+
+    /**
+     * Writes event to the registered logging providers.
+     */
+    public logEvent(
+        eventMessage: string,
+        properties: { [name: string]: string },
+        measurements?: any) {
+        if (this.eventPropertiesLocalCopy) {
+            for (let id in this.eventPropertiesLocalCopy) {
+                properties[id] = String(this.eventPropertiesLocalCopy[id]);
+            }
+        }
+
+        if (this.commonDetectorEventProperties) {
+            for (let id in this.commonDetectorEventProperties) {
+                properties[id] = String(this.commonDetectorEventProperties[id]);
+            }
+        }
+
+        try {
+            for (var telemetryProvider of this.telemetryProviders) {
+                telemetryProvider.logEvent(eventMessage, properties, measurements);
+                for (let iter in properties) {
```

Comment:
I guess you added this for testing purposes. Remove the console messages before checking-it in for PROD or only log to console if the environment is dev

Reply:
Thanks for spotting that :) Removed the console log.